spec/helpers/shf_applications_helper_spec.rb

```diff
@@ -2,6 +2,38 @@
 
 RSpec.describe ShfApplicationsHelper, type: :helper do
 
+
+  describe '#states_selection_list gets the localized version of each state name each time it is requested' do
+
+    around(:each) do |example|
+      orig_locale = I18n.locale
+      example.run
+      I18n.locale = orig_locale
+    end
+
+    let(:application) { create(:shf_application) }
+
+    it 'returns the list in the default I18n locale' do
+      select_list = helper.states_selection_list
+      select_list.each do |each_option|
+        expect(each_option[0]).to eq I18n.t("activerecord.attributes.shf_application.state/#{each_option[1]}")
+      end
+    end
+
+    it 'correct locale if changed to :en from locale = :sv' do
+      orig_locale = I18n.locale
+      I18n.locale = :sv
+      helper.states_selection_list
+
+      I18n.locale = :en
+      select_list = helper.states_selection_list
+      select_list.each do |each_option|
+        expect(each_option[0]).to eq I18n.t("activerecord.attributes.shf_application.state/#{each_option[1]}", locale: :en)
+      end
+      I18n.locale = orig_locale
```

Comment:
ditto

Reply:
_it's clear how I arrived at the "gee - I should use `around(...)` approach"_

Reply:
Yes, been there done that :)